chicago/events.py

```diff
@@ -29,40 +31,86 @@ def eventPages(self, event_url, search_type='all') :
 
-                status_string = location_list[-1].split('Chicago, Illinois')
-                if len(status_string) > 1 and status_string[1] :
-                    status = status_string[1].lower()
-                    if status not in ['cancelled', 'tentative', 'confirmed', 'passed'] :
-                        print(status)
-                        status = 'confirmed'
-                else :
-                    status = 'confirmed'
-
-
-
                 when = events[u'Meeting\xa0Date']
                 time_string = events[u'Meeting\xa0Time']
                 event_time = datetime.datetime.strptime(time_string,
                                                         "%I:%M %p")
                 when = when.replace(hour=event_time.hour)
 
+                status_string = location_list[-1].split('Chicago, Illinois')
+                if len(status_string) > 1 and status_string[1] :
+                    status_text = status_string[1].lower()
+                    if any(phrase in status_text 
+                           for phrase in ('rescheduled to',
+                                          'postponed to',
+                                          'reconvened to',
+                                          'recessed',
+                                          'cancelled',
+                                          'new date and time',
+                                          'rescheduled indefinitely',
+                                          'rescheduled for')) :
+                        status = 'cancelled'
+                    elif status_text in ('rescheduled') :
+                        status = 'cancelled'
+                    else :
+                        print(status_text)
+                elif datetime.datetime.utcnow().replace(tzinfo = pytz.utc) > when :
+                    status = 'confirmed'
+                else :
+                    status = 'passed'
+                            
+
                 e = Event(name=events["Name"]["label"],
-                          when=when,
+                          start_time=when,
+                          timezone='US/Central',
                           location=location,
                           status=status)
-                e.add_source(EVENTSPAGE)
-                if events['Video'] != u'Not\xa0available' :
-                    print(events['Video'])
+                e.add_source(detail_url)
+                if events['Video'] != 'Not\xa0available' : 
+                    e.add_media_link(note='Recording',
+                                     url = events['Video']['url'],
+                                     type="recording",
+                                     media_type = '???')
+
+                addDocs(e, events, 'Agenda')
+                addDocs(e, events, 'Notice')
+                addDocs(e, events, 'Transcript')
+                addDocs(e, events, 'Summary')
+
+                if events["Name"]["label"] != "City Council" :
```

Comment:
I appreciate that, but if we turn this on it will mean we have to scrape all the bills in order to get through the events. Let's hold off on doing that until we have figure out better update strategy for legistar systems.